checkout/domain/sourcingengine.go

```diff
@@ -112,18 +113,23 @@ func (se *SourcingEngine) SetSourcesForCartItems(ctx context.Context, session *w
 		return nil
 	}
 	for _, decoratedDelivery := range decoratedCart.DecoratedDeliveries {
+		itemSources := make(map[string]string)
 		for _, decoratedCartItem := range decoratedDelivery.DecoratedItems {
 			sourceID, err := se.SourcingService.GetSourceID(ctx, session, decoratedCart, decoratedDelivery.Delivery.DeliveryInfo.Code, &decoratedCartItem)
 			if err != nil {
 				se.Logger.WithContext(ctx).WithField("subcategory", "SourcingEngine").Error(err)
 				return fmt.Errorf("checkout.application.sourcingengine error: %v", err)
 			}
 			se.Logger.WithContext(ctx).WithField("category", "checkout").WithField("subcategory", "SourcingEngine").Debug("SourcingEngine detected source %v for item %v", sourceID, decoratedCartItem.Item.ID)
-			err = se.Cartservice.UpdateItemSourceID(ctx, session, decoratedCartItem.Item.ID, decoratedDelivery.Delivery.DeliveryInfo.Code, sourceID)
-			if err != nil {
-				return errors.Wrap(err, "Could not update cart item")
-			}
+
+			itemSources[decoratedCartItem.Item.ID] = sourceID
+		}
+
+		err := se.Cartservice.UpdateItemsSourceID(ctx, session, decoratedDelivery.Delivery.DeliveryInfo.Code, itemSources)
```

Comment:
If change suggested above is done we can collect all update commands and just have one call